spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/outbound/Mqttv5PahoMessageHandler.java

```diff
@@ -152,17 +167,25 @@ protected void onInit() {
 	@Override
 	protected void doStart() {
 		try {
-			this.mqttClient.connect(this.connectionOptions).waitForCompletion(getCompletionTimeout());
+			var clientManager = getClientManager();
+			if (this.mqttClient != null) {
+				this.mqttClient.connect(this.connectionOptions).waitForCompletion(getCompletionTimeout());
+			}
+			else if (clientManager != null) {
+				this.mqttClient = clientManager.getClient();
+			}
 		}
 		catch (MqttException ex) {
-				logger.error(ex, "MQTT client failed to connect.");
+			logger.error(ex, "MQTT client failed to connect.");
 		}
 	}
 
 	@Override
 	protected void doStop() {
 		try {
-			this.mqttClient.disconnect().waitForCompletion(getDisconnectCompletionTimeout());
+			if (this.mqttClient != null) {
```

Comment:
Since you do now:
```
			else if (clientManager != null) {
				this.mqttClient = clientManager.getClient();
			}
```
in the `doStart()`, we have to check here for the `getClientManager() != null` before performing `disconnect()`.
As we agreed: we cannot not close the client we don't owe.
Same applies for the `close()` later on in the `destroy()`

Reply:
You are right - changed it at the last moment and missed this part

Reply:
I wonder if the `mqttClient != null` check is redundant in `doStart`/`doStop`/`destroy` methods. It cannot be null from `onInit` method - if client cannot be created we throw `BeanCreationException`